src/workerd/util/sqlite.h

```diff
@@ -20,6 +20,15 @@ namespace workerd {
 using kj::byte;
 using kj::uint;
 
+// Used to collect periodic metrics about queries and size of sqlite db
+class SqliteObserver {
+public:
+  virtual void updateQueryStats(uint64_t rowsRead, uint64_t rowsWritten) {}
+  virtual void setSrsStoredBytes(uint64_t srsStoredBytes) {}
+
+  static SqliteObserver DEFAULT;
```

Comment:
Could this be `constexpr` like `SqliteDatabase::TRUSTED`?

Reply:
This can't be a constexpr because of the virtual methods in it. 
It can't be a const either because the methods will eventually update the state in the implementation of the derived class